server/ai_mediaserver.go

```diff
@@ -255,20 +255,18 @@ func (ls *LivepeerServer) LLM() http.Handler {
 		requestID := string(core.RandomManifestID())
 		ctx = clog.AddVal(ctx, "request_id", requestID)
 
-		var req worker.GenLLMFormdataRequestBody
-
-		multiRdr, err := r.MultipartReader()
-		if err != nil {
+		var req worker.GenLLMJSONRequestBody
+		if err := jsonDecoder(&req, r); err != nil {
 			respondJsonError(ctx, w, err, http.StatusBadRequest)
 			return
 		}
 
-		if err := runtime.BindMultipart(&req, *multiRdr); err != nil {
-			respondJsonError(ctx, w, err, http.StatusBadRequest)
+		if req.Model == nil || req.Messages == nil || req.Stream == nil || req.MaxTokens == nil {
```

Comment:
I guess we can close https://github.com/livepeer/go-livepeer/pull/3204?

Reply:
Yes we can. I tested removing all these items from the request and confirmed did not seg fault.

Also added a check that `req.Messages` is not empty because it would seg fault the worker if no messages were sent in the request.